### PR TITLE
VPS-60/reset button fix

### PR DESCRIPTION
**Pull request by codecreator127**
## Describe the issue

Reset button wasn't working correctly
https://wdcc-vps.atlassian.net/jira/software/projects/VPS/boards/2?selectedIssue=VPS-60

## Describe the solution

Separated the delete all flags and delete all paths into separate functions:

image
image

## Risk

May break the reset functionality

## Definition of Done

- [ ] Code peer-reviewed
- [ ] Wiki Documentation is written and up to date
- [ ] Unit tests written and passing
- [ ] Integration tests written and passing
- [ ] Continuous Integration build passing
- [ ] Acceptance criteria met
- [ ] Deployed to production environment

## Reviewed By

Who reviewed your PR - for commit history once merged

### CODE REVIEW

**Comment by JordanBlenn** [comment]
Needs to run prettier

File: backend/src/routes/api/navigate/group.js [hunk 1]
```diff
@@ -40,6 +40,20 @@ const deleteAllNotes = async (groupData) => {
...
 
+const deleteAllFlags = async (groupData) => {
+  const groupId = groupData._id;
+  const group = await Group.findById(groupId).lean();
+  if (!group) {
+    throw new HttpError("Group not found", STATUS.NOT_FOUND);
+  }
+
+  await Group.updateOne({ _id: groupId }, { $set: { currentFlags: [] } }).exec();
+  // if (res.nModified !== 1) {
```
Comment (wjin-lee) [inline backend/src/routes/api/navigate/group.js:51]: should this be here?

File: backend/src/routes/api/navigate/group.js [hunk 2]
```diff
@@ -198,15 +215,20 @@ export const groupReset = async (req) => {
...
   const hasReset = scene.components.some((c) => c.type === "RESET_BUTTON");
   if (!hasReset) throw new HttpError("Invalid reset", STATUS.FORBIDDEN);
 
-  await Group.findOneAndUpdate(
-    { _id: group._id },
-    { $set: { path: [], currentFlags: [] } },
-    { new: true }
-  );
 
+  console.log(group)
```
Comment (wjin-lee) [inline backend/src/routes/api/navigate/group.js:227]: potentially redundant console.log

File: backend/src/routes/api/navigate/group.js [hunk 3]
```diff
@@ -221,6 +243,8 @@ export const groupGetResources = async (req) => {
   const flags = group.currentFlags || [];
   const resources = [];
 
+  console.log(flags)
```
Comment (wjin-lee) [inline backend/src/routes/api/navigate/group.js:246]: potentially redundant console.log

**Review by wjin-lee — CHANGES_REQUESTED** [review]
Nice work - mostly nits except for the failing CI/CD items!

**Review by wjin-lee — APPROVED** [review]
lgtm!

**Pull request merged**